streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -361,12 +637,94 @@ void writeAsText(final String filePath,
      * The store name must be a valid Kafka topic name and cannot contain characters other than ASCII alphanumerics, '.', '_' and '-'.
      *
      * @param topic     the topic name
-     * @param storeName the state store name used for the result {@code KTable}; valid characters are ASCII
-     *                  alphanumerics, '.', '_' and '-'
+     * @param queryableStoreName the state store name used for the result {@code KTable}; valid characters are ASCII
+     *                  alphanumerics, '.', '_' and '-'. If {@code null} this is the equivalent of {@link KTable#through(String)()}
+     * @return a {@code KTable} that contains the exact same (and potentially repartitioned) records as this {@code KTable}
+     */
+    KTable<K, V> through(final String topic,
```

Comment:
I understand adding the store name / store supplier here is to achieve consistency with `builder#table`, but I'm wondering what would be the recommended pattern for using through. I.e. with the following topology: 

```
KTable table1 = table0.someOps(..., "name1");
KTable table2 = table1.through("topic", "name2");
```

What's the difference of querying store `name1` and querying store `name2`? Would those be the same stores? Shall we in this case save one store in the underlying impl?